### PR TITLE
UI overhaul

### PR DESCRIPTION
**Pull request by MadmanMartian**
Overhauls the UI

Removes the right ear slot, concatenates them into a single ears slot

Adds the sling slot, for putting a weapon over your shoulder.

Fields of Fire - Undisclosed Location 2019-06-01 171813

### CODE REVIEW

File: code/game/objects/items/weapons/grenades/flashbang.dm [hunk 1]
```diff
@@ -37,7 +37,7 @@
 		if(iscarbon(M))
 			eye_safety = M.eyecheck()
 			if(ishuman(M))
-				if(istype(M:l_ear, /obj/item/clothing/ears/earmuffs) || istype(M:r_ear, /obj/item/clothing/ears/earmuffs))
+				if(istype(M:ears, /obj/item/clothing/ears/earmuffs))
```
Comment (Quantomicus) [inline code/game/objects/items/weapons/grenades/flashbang.dm]: This has always been gross and will continue to be. A landmark of SS13 code.

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -773,49 +773,48 @@
...
-									health_images += image('icons/mob/screen1_health.dmi',"softcrit")
-								if(trauma_val >= 1)
-									health_images += image('icons/mob/screen1_health.dmi',"hardcrit")
-						else if(no_damage)
-							health_images += image('icons/mob/screen1_health.dmi',"fullhealth")
-
 						healths.overlays += health_images
 
+		if(healthbar)
+			var/phys_health = maxHealth - getOxyLoss() - getToxLoss() - getFireLoss() - getBruteLoss() - getCloneLoss()
```
Comment (Quantomicus) [inline code/modules/mob/living/carbon/human/life.dm]: Is there no centralized solution to get true health?

File: code/modules/projectiles/projectile/energy.dm [hunk 3]
```diff
@@ -136,7 +136,7 @@
 	var/mob/living/carbon/human/H = M
 	if(iscarbon(M))
 		if(ishuman(M))
-			if(istype(H.l_ear, /obj/item/clothing/ears/earmuffs) || istype(H.r_ear, /obj/item/clothing/ears/earmuffs))
+			if(istype(H.ears, /obj/item/clothing/ears/earmuffs))
```
Comment (Quantomicus) [inline code/modules/projectiles/projectile/energy.dm]: Oof. Another one.

File: code/game/objects/items/weapons/grenades/flashbang.dm (hunk 1 above)
Comment (MadmanMartian) [reply code/game/objects/items/weapons/grenades/flashbang.dm]: I'll make a quick macro for ear protection

**Comment by TheBimmer231** [comment]
raise hunger, thirst, oxygen and temperature indicators by a single pixel 
image

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -235,6 +235,9 @@
 /mob/living/carbon/proc/eyecheck()
 	return 0
 
+/mob/living/carbon/proc/earcheck()
```
Comment (Quantomicus) [inline code/modules/mob/living/carbon/carbon.dm]: Ummmm, didn't you just completely disregard helmets, earmuffs and such?

Comment (MadmanMartian) [reply code/modules/mob/living/carbon/carbon.dm]: This is just a carbon definition to mimic the eyecheck. ear_protection is supposed to be an override of this for humans. Good spot, fixing now.

File: code/modules/mob/living/carbon/human/life.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/modules/mob/living/carbon/human/life.dm]: Unfortunately not, as health seems to be based on the brain for this codebase, whereas this health bar is in regard to the body health.

File: code/modules/mob/living/carbon/human/human.dm [hunk 5]
```diff
@@ -659,7 +659,7 @@
 		return FLASH_PROTECTION_MAJOR
 	return total_protection
 
-/mob/living/carbon/human/proc/ear_protection()
+/mob/living/carbon/human/earcheck()
```
Comment (MadmanMartian) [inline code/modules/mob/living/carbon/human/human.dm]: @Quantomicus sorted

**Pull request merged**